### DISCUSSION TITLE
Why is NodeJS so restricted?

### DISCUSSION THREAD

**Discussion by TheBrenny** [gluesql/gluesql#1356] [Ideas]
I would expect the functionality to be identical if a package like `napi-bindgen` or `neon` were used, but apparently, the NodeJS package is the most restricted...

Was there a design decision that lead to separating the Node bindings from the core rust, or was it part of the todo list? I'm really curious and would be keen to help, but I don't know enough rust by a long shot! Especially when it comes to makinging FFI bindings (even with Neon!).

**Comment by panarch** [comment]
currently nodejs package is built using wasm-pack so it only supports in-memory storage. but yes, as you mentioned, if we use napi or neon, then we will be able to provide almost full feature support as we do for rust package.
it's just because our resource is limited. though we have plan to add support for nodejs but we cannot ensure the date.
so we always welcome for new contributors to participate on this :)

**Reply by TheBrenny** [reply]
[Challenge accepted?](https://github.com/TheBrenny/gluesql-bindings)

I imagine it'd be easier to wrap GlueSQL, but I also imagine it'll come at a performance cost...
